src/main/java/hudson/plugins/brakeman/BrakemanResult.java

```diff
@@ -89,6 +116,118 @@ public String getDisplayName() {
         return Messages.Brakeman_ProjectAction_Name();
     }
 
+    public void setReason(String reason) {
+        this.reason = reason;
+    }
+
+    @Override
+    public String getReason() {
+        return reason;
+    }
+
+    @SuppressWarnings("hiding")
+    @Override
+    public void evaluateStatus(final Thresholds thresholds, final boolean useDeltaValues, final PluginLogger logger, final String url) {
+        evaluateStatus(thresholds, useDeltaValues, true, logger, url);
+    }
+
+    /**
+     * Updates the build status, i.e. sets this plug-in result status field to
+     * the corresponding {@link Result}. Additionally, the {@link Result} of the
+     * build that owns this instance of {@link BuildResult} will be also
+     * changed.
+     *
+     * This has been overridden to add a different evaluate strategy that
+     * evaluates thresholds against non-ignored warnings/annotations.
+     *
+     * @param thresholds
+     *            the failure thresholds
+     * @param useDeltaValues
+     *            the use delta values when computing the differences
+     * @param canComputeNew
+     *            determines whether new warnings should be computed (with
+     *            respect to baseline)
+     * @param logger
+     *            the logger
+     * @param url
+     *            the URL of the results
+     */
+    // CHECKSTYLE:OFF
+    @SuppressWarnings("hiding")
+    @Override
+    public void evaluateStatus(final Thresholds thresholds, final boolean useDeltaValues, final boolean canComputeNew,
+                               final PluginLogger logger, final String url) {
+        // CHECKSTYLE:ON
+        if(isSuccessful()) {
+            this.thresholds = thresholds;
+            this.useDeltaValues = useDeltaValues;
+
+            BuildResultEvaluator resultEvaluator = new BuildResultEvaluator(url);
+            Result buildResult;
+            StringBuilder messages = new StringBuilder();
+            Set<FileAnnotation> annotations = getNonIgnoredAnnotations();
```

Comment:
so non-ignored annotations are the only ones that can fail builds for the time being